apps/src/code-studio/components/AdvancedShareOptions.jsx

```diff
@@ -141,7 +142,13 @@ class AdvancedShareOptions extends React.Component {
           exporting: false,
           exportError: 'Failed to export project. Please try again later.'
         });
-      });
+    });
+    firehoseClient.putRecord({
+      study: 'applab',
+      study_group: 'share_options',
+      event: 'download_export',
+      data_string: exporting ? 'exported successfully' : 'export failed'
```

Comment:
there are a couple of issues here -- 
- `exporting` isn't in scope -- it's part of our react state, so if we want to access it's value, we have to use `this.state.exporting`
- however, we can't just change this line to `this.state.exporting` -- its value will always be `true` because the function above (`this.props.exportApp()`) is asynchronous. we need to log this firehose event from within the callbacks of that function to determine whether or not the export was successful.

we probably want something like this:

```javascript
logExportEvent = successful => {
  firehoseClient.putRecord({
      study: 'applab',
      study_group: 'share_options',
      event: 'download_export',
      data_string: successful ? 'exported successfully' : 'export failed'
  });
}

downloadExport = () => {
  this.setState({exporting: true});
  const onSuccess = () => {
    this.setState({exporting: false});
    this.logExportEvent(true);
  };
  const onFailure = () => {
    this.setState({exporting: false, exportError: 'Failed to export project. Please try again later.'});
    this.logExportEvent(false);
  };

  this.props
    .exportApp()
    .then(onSuccess, onFailure);
}
```

caveat: i haven't tested this code 😅

Reply:
Thanks Maddie! This is a bit beyond my comfort level then so I will move this task to up next for an engineer to add the firehose event - they can use this PR discussion to get a head start :)